components/LocalePicker.php

```diff
@@ -142,7 +150,15 @@ protected function makeLocaleUrlFromPage($locale)
             $router = new RainRouter;
 
             $params = $this->getRouter()->getParameters();
-
+            
+            // Provide an opportunity to manipulate the parameters
```

Comment:
Needs inline documentation for the event, see https://github.com/octobercms/october/blob/master/modules/system/console/OctoberMirror.php#L96-L108